solve4x4Cube.scala

```diff
@@ -0,0 +1,196 @@
+import scala.math.{Pi, cos, sin}
```

Comment:
Could you add Apache Software LIcense 2 copyright header on all the files (if you are OK with licensing this way)?  It's a generous open source license.

Reply:
Yes, I am adding it.